common/cmake/devices.cmake

```diff
@@ -963,6 +963,13 @@ function(ADD_FPGA_TARGET)
     if(NOT "${ADD_FPGA_TARGET_INPUT_IO_FILE}" STREQUAL "")
       add_file_target(FILE ${ADD_FPGA_TARGET_INPUT_IO_FILE})
     endif()
+    if(NOT "${ADD_FPGA_TARGET_INPUT_XDC_FILE}" STREQUAL "")
+      add_file_target(FILE ${ADD_FPGA_TARGET_INPUT_XDC_FILE})
+      get_file_location(INPUT_XDC_FILE ${ADD_FPGA_TARGET_INPUT_XDC_FILE})
+      if(NOT "${ADD_FPGA_TARGET_PART}" STREQUAL "")
+        set(PART_JSON ${PRJXRAY_DB_DIR}/${ARCH}/${ADD_FPGA_TARGET_PART}.json)
```

Comment:
Get this from the `${BOARD}` target.  Maybe 
`get_property_required(PART_JSON ${BOARD} PART_JSON)`

and add the PART_JSON here: https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/xc7/make/device_define.cmake#L56

Reply:
Done